backend/tracim_backend/lib/search/es_models.py

```diff
@@ -0,0 +1,142 @@
+from fnmatch import fnmatch
+import os
+import typing
+
+from elasticsearch_dsl import Boolean
+from elasticsearch_dsl import Date
+from elasticsearch_dsl import Document
+from elasticsearch_dsl import InnerDoc
+from elasticsearch_dsl import Integer
+from elasticsearch_dsl import Keyword
+from elasticsearch_dsl import Nested
+from elasticsearch_dsl import Object
+from elasticsearch_dsl import Text
+from elasticsearch_dsl import analysis
+from elasticsearch_dsl import analyzer
+
+# INFO - G.M - 2019-05-31 - Analyzer/indexing explained:
+# Instead of relying of wildcard for autocompletion which is costly and make some feature doesn't
+# work correctly, for example ranking, We use ngram mecanism.
+# This means that for work "elephant", we will index thing like "ele", "elep", "lepha", etc...
+# As we don't want to have a *text* matching but only an autocomplete matching like text*, we use
+# edge_ngram, we will only index for "elephant": "ele" , "elep" , "eleph" , etc..
+# We want that ele match elephant result, but we do not want that elephant match ele result,
+# that's why we set different analyzer for search (we search word given) and indexing (we index ngram
+# of label of content to allow autocompletion)
+
+
+edge_ngram_token_filter = analysis.token_filter(
+    "edge_ngram_filter", type="edge_ngram", min_ngram=2, max_gram=20
+)
+# INFO - G.M - 2019-05-23 - search_analyser: do search for content given an some similar word
+folding = analyzer("folding", tokenizer="standard", filter=["lowercase", "asciifolding"])
+# INFO - G.M - 2019-05-23 -  index_analysers, index edge ngram for autocompletion and strip html for indexing
+edge_ngram_folding = analyzer(
+    "edge_ngram_folding",
+    tokenizer="standard",
+    filter=["lowercase", "asciifolding", edge_ngram_token_filter],
+)
+html_folding = analyzer(
+    "html_folding",
+    tokenizer="standard",
+    filter=["lowercase", "asciifolding", edge_ngram_token_filter],
+    char_filter="html_strip",
+)
+
+DEFAULT_INDEX_DOCUMENTS_ALIAS = "tracim_contents"
+DEFAULT_INDEX_DOCUMENTS_PATTERN_TEMPLATE = "{index_alias}-{date}"
+# FIXME - G.M - 2019-05-24 - hack using env var only configuration of document alias and document
+# pattern now, as it's currently complex to rewrite Lib signature to get needed information (we
+# need to convert static IndexedContent object to something dynamic that we will pass to SearchApi)
+# this should be refactor to have a true tracim config parameter for this.
+INDEX_DOCUMENTS_ALIAS = os.environ.get(
+    "TRACIM_SEARCH__ELASTICSEARCH__INDEX_ALIAS", DEFAULT_INDEX_DOCUMENTS_ALIAS
+)
+INDEX_DOCUMENTS_PATTERN_TEMPLATE = os.environ.get(
+    "TRACIM_SEARCH__ELASTICSEARCH__INDEX_PATTERN_TEMPLATE", DEFAULT_INDEX_DOCUMENTS_PATTERN_TEMPLATE
+)
+INDEX_DOCUMENTS_PATTERN = INDEX_DOCUMENTS_PATTERN_TEMPLATE.replace("{date}", "*").replace(
+    "{index_alias}", INDEX_DOCUMENTS_ALIAS
+)
```

Comment:
You can use `str.format` instead replace:

``` python
>>> MY_STR = "Hello, {word}!"
>>> MY_STR.format(word="little cat")
'Hello, little cat!'
```

Reply:
Done.